pkg/clients/utils.go

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package clients
+
+import (
+	"fmt"
+	"regexp"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type Cmder interface {
+	GetCommand() []string
+	ExtractResult(string) (map[string]string, error)
+}
+
+type Cmd struct {
+	key         string
+	prefix      []string
+	suffix      []string
+	cmd         []string
+	cleanupFunc func(string) string
+	regex       *regexp.Regexp
+	fullCmd     []string
+}
+
+func NewCmd(key string, cmd []string) (*Cmd, error) {
```

Comment:
Does it actually matter that the key is? I think it can be handled internally to the `Cmd`

Reply:
Yes the key relates back to struct tags.